src/ray/object_manager/plasma/create_request_queue.cc

```diff
@@ -173,12 +196,31 @@ Status CreateRequestQueue::ProcessRequests() {
         oom_start_time_ns_ = now;
       }
 
-      // Notify the scheduler that object creation is blocked at this priority.
-      on_object_creation_blocked_callback_(queue_it->first.first);
-      if (!should_spill_) {
-        RAY_LOG(INFO) << "Object creation of priority " << queue_it->first.first << " blocked";
-        return Status::TransientObjectStoreFull("Waiting for higher priority tasks to finish");
-      }
+	  if(RayConfig::instance().enable_BlockandEvictTasks()){
+		on_object_creation_blocked_callback_(queue_it->first.first);
+		on_object_evict_callback_(queue_it->first.first);
+        if (!should_spill_) {
+          RAY_LOG(INFO) << "Object creation of priority " << queue_it->first.first << " blocked";
+          return Status::TransientObjectStoreFull("Waiting for higher priority tasks to finish");
+        }
+		RAY_LOG(INFO) << "[JAE_DEBUG] should_spill set";
+			    SetShouldSpill(false);
```

Comment:
I don't think we should set should_spill_ here, right? We should allow the callbacks to set this later on. What was the reason for setting it?